events/views.py

```diff
@@ -12,24 +14,27 @@
 logger = logging.getLogger(__name__)
 
 
-class EventView(TemplateView):
-    ''' Show single Event '''
-    template_name = 'events/event-detail.html'
+class EventDetailView(DetailView):
+    model = Event
+    context_object_name = 'event'
 
-    def get_context_data(self, **kwargs):
-        context = super(EventView, self).get_context_data(**kwargs)
-        context['event'] = get_object_or_404(Event, pk=kwargs['pk'], published=True)
-        return context
 
+class EventsListView(ListView):
+    model = Event
+    context_object_name = 'events'
 
-class EventsView(TemplateView):
-    ''' List all Events '''
-    template_name = 'events/events.html'
+    def get_queryset(self):
+        return Event.objects.filter(published=True, start__gte=timezone.now())
 
-    def get_context_data(self, **kwargs):
-        context = super(EventsView, self).get_context_data(**kwargs)
-        context['events'] = Event.objects.filter(published=True, start__gte=timezone.now())
-        return context
+
+class LocationDetailView(DetailView):
+    model = Location
+    context_object_name = 'location'
+
+
+class LocationsListView(ListView):
+    model = Location
```

Comment:
I don't know a lot about the `ListView` base class. What ensures the location list comes back in a consistent (and useful) order? Or the items going to be sorted by key by default? Should we at least sort by name?

Reply:
Yes, the model is setup to sort by name: https://github.com/pyatl/big-peach/blob/develop/events/models.py#L28

Reply:
Perfect